test/src/tests-identity.ts

```diff
@@ -307,59 +437,85 @@ describe('identity', function() {
         window.mParticle.config.kitConfigs.push(config1);
         window.mParticle.config.kitConfigs.push(config2);
 
-        mockServer.respondWith(urls.identify, [
-            200,
-            {},
-            JSON.stringify({ mpid: 'MPID1', is_logged_in: false }),
-        ]);
+        fetchMockSuccess(urls.identify, {
+                    mpid: 'MPID1',
+                    is_logged_in: false,
+        });
 
         mParticle.init(apiKey, window.mParticle.config);
 
-        let activeForwarders = mParticle.getInstance()._getActiveForwarders();
+        waitForCondition(hasIdentifyReturned)
+            .then(() => {
+                let activeForwarders = mParticle
+                    .getInstance()
+                    ._getActiveForwarders();
         expect(activeForwarders.length).to.equal(0);
 
         let consentState = mParticle.Consent.createConsentState();
         consentState.addGDPRConsentState(
             'foo purpose 1',
             mParticle.Consent.createGDPRConsent(true)
         );
-        mParticle.Identity.getCurrentUser().setConsentState(consentState);
+                mParticle.Identity.getCurrentUser().setConsentState(
+                    consentState
+                );
 
-        activeForwarders = mParticle.getInstance()._getActiveForwarders();
+                activeForwarders = mParticle
+                    .getInstance()
+                    ._getActiveForwarders();
         activeForwarders.length.should.equal(1);
         activeForwarders[0].name.should.equal('MockForwarder1');
 
-        mockServer.respondWith(urls.login, [
-            200,
-            {},
-            JSON.stringify({ mpid: 'MPID2', is_logged_in: false }),
-        ]);
+                // TODO: Make a convenience method for this
+                fetchMock.post(
```

Comment:
follow this which is what i did when you made the fetchMockSuccess utility function

https://github.com/mParticle/mparticle-web-sdk/pull/921/files#diff-5e2ebcc7bc661f0ac56f35e749148d68843f2db99f22f3bfb44e5b950ce8fe2fR954-R956